.github/actions/setup-go-cache/action.yml

```diff
@@ -0,0 +1,37 @@
+name: "Setup Go"
+description: "Setup Go with caching"
+inputs:
+  cache-prefix:
+    description: "Prefix for the cache key"
+    required: false
+    default: "go"
+  disable-cache:
+    description: "Disable caching"
+    required: false
+    default: "false"
+  go-version:
+    description: "Version of Go. Default 1.21"
+    required: false
+    default: "1.21"
```

Comment:
We should ask the GR guys some time if they have any interest in updating the `armada` repo to Go 1.21. (I've been building Armada with 1.21.1 and it seems to work fine).